regexlint/checkers.py

```diff
@@ -506,14 +469,18 @@ def check_redundant_repetition(reg, errs):
 def manual_check_for_empty_string_match(reg, errs, raw_pat):
     # Note: callback functions get a pass here, since they're used for
     # indentation tracking in SassLexer (and friends).
-    # Explicitly, '#pop' and 'next-state' ARE checked because if they
-    # intentionally match on empty string, they should be using default().
-    if isinstance(raw_pat[1], Token.__class__):
-        regex = re.compile(raw_pat[0])
-        # Either match on empty string, or empty string at the end of a word
-        if regex.match('') or regex.match('a', 1):
-            errs.append(('999', logging.ERROR, 0, 'Matches empty string'))
-        #remove_error(errs, '103')
+    # '#pop' and 'next-state' ARE checked if pattern is empty because they
+    # should be using default().
+    if not isinstance(raw_pat[1], Token.__class__):
+        return
+    if raw_pat[0] != '' and len(raw_pat) > 2:
```

Comment:
Could you include a couple of lines of comment, I don't follow this change.  Is this for the callback functions mentioned a few lines above?

Reply:
It's for patterns that effect a state change with a zero-width match.  Something like `(r"(?=xyz)", Text, "#pop")`, I don't think there's a better way to write it, and I don't think it's harmful since it will not lead to infinite looping in the same state.

I'll move the comments around a bit and clarify them.